subprojects/plugin-use/src/integTest/groovy/org/gradle/plugin/use/PluginUseClassLoadingIntegrationSpec.groovy

```diff
@@ -68,18 +68,15 @@ class PluginUseClassLoadingIntegrationSpec extends AbstractIntegrationSpec {
         succeeds("verify")
     }
 
-    def "plugin class isn't visible to build script"() {
+    def "plugin class is visible to build script"() {
         publishPlugin()
 
         buildScript """
             $USE
 
             task verify {
                 doLast {
-                    try {
-                        getClass().getClassLoader().loadClass("org.gradle.test.TestPlugin")
-                        throw new AssertionError("plugin class *is* visible to build script")
-                    } catch (ClassNotFoundException expected) {}
+                    getClass().getClassLoader().loadClass("org.gradle.test.TestPlugin")
```

Comment:
That's because the `PluginResolutionServiceTestServer` now returns `legacy = true` (see https://github.com/gradle/gradle/blob/381363a88958c209a0097d5a5393b8c0167133f9/subprojects/plugin-use/src/testFixtures/groovy/org/gradle/plugin/use/resolve/service/PluginResolutionServiceTestServer.groovy#L109) so we would get more coverage from tests on the class loading aspects we want to preserve.